apps/frontend/src/libs/components/icon/icon.tsx

```diff
@@ -0,0 +1,27 @@
+/// <reference types="vite-plugin-svgr/client" />
+
+import { type FC } from "react";
+
+import Eye from "~/assets/images/icons/eye.svg?react";
+import StrikedEye from "~/assets/images/icons/striked-eye.svg?react";
+
+type IconName = "eye" | "strikedEye";
+
+const iconNameToSvg: Record<IconName, FC<React.SVGProps<SVGSVGElement>>> = {
+	eye: Eye,
+	strikedEye: StrikedEye,
+};
+
+interface Properties {
+	className?: string | undefined;
+	name: IconName;
+	onClick?: () => void;
+}
+
+const Icon: FC<Properties> = ({ className, name, onClick }: Properties) => {
```

Comment:
Other components declared in the following way: `const Component = ({ ... }: Properties): JSX.Element => {` so I think it's better to do the same for your `Icon` to keep code consistent